### PR TITLE
lab1 commit idk

### PR DESCRIPTION
**Pull request by auvy**

### CODE REVIEW

File: lab1/main.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from src.map import Map
+
+input_path             = './data/input.txt'
+output_path            = './data/output.txt'
+
+def app():
+  case_list = []
+  try:
+    case_list = parse_input(input_path)
+    # for c in case_list:
```
Comment (svtxvt) [inline lab1/main.py]: Remove comments pls

File: lab1/src/map.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    # class-private
+    def __check_neighbors_countries(self):
+        if len(self.countries) < 2:
+            return
+        for country in self.countries:
+            if not country.has_foreign_neighbors():
+                raise Exception(f"{country.name} has no connection with other countries")
+
+
+# not class methods but needed to check validity only
```
Comment (svtxvt) [inline lab1/src/map.py]: Maybe we can move it to a separate module?

File: lab1/src/map.py [hunk 3]
```diff
@@ -71,7 +72,7 @@ def diffuse(self):
             day += 1
 
         # self.countries.sort()
-        self.countries.sort(key=lambda x: x.completion_day, reverse=False)
+        self.countries.sort(key=lambda x: (x.completion_day, x.name))
```
Comment (auvy) [inline lab1/src/map.py]: added secondary sorting by name, as required per task

File: lab1/src/map.py [hunk 4]
```diff
@@ -95,32 +96,4 @@ def __check_neighbors_countries(self):
...
-  for country in case:
-    coords = list(country.values())
-    coords.pop(0)
-    for c in coords:
-      if int(c) <= 0 or int(c) >= (grid_size + 1):
-        raise Exception(f'Error: Country "{country["name"]}" coordinates out of bounds.')
-  
-  if len(case) > 1:
-    check_bounds(case)
-
```
Comment (auvy) [inline lab1/src/map.py]: moved functions into separate 'validation' module

**Pull request merged**